index.js

```diff
@@ -33,15 +33,15 @@ function HyperdriveHttp (getArchive) {
     var key = archive
       ? encoding.encode(archive.key)
       : segs.shift()
-    var filename = segs.shift()
+    var filename = segs.join('/')
     var op = 'get'
 
     try {
       // check if we are serving archive at root
       key = key.replace(/\.changes$/, '')
-      encoding.decode(key)
+      encoding.encode(Buffer.from(key, 'hex'))
```

Comment:
isn't this what `dat-encoding` is already doing? https://github.com/juliangruber/dat-encoding/blob/master/index.js#L12